devito/ir/clusters/cluster.py

```diff
@@ -362,6 +362,13 @@ def dspace(self):
         intervals = intervals.promote(lambda d: not d.is_Sub)
         intervals = intervals.zero(set(intervals.dimensions) - oobs)
 
+        # Upper bound of intervals including dimensions classified for
```

Comment:
This comment doesn't sound too clear. I think shorter + tiny example might help? 

perhaps this is just me nitpicking...